js/src/Label.js

```diff
@@ -65,79 +64,95 @@ define(["d3", "./Mark"], function(d3, MarkViewModule) {
-            if(this.x_offset !== undefined || this.y_offset !== undefined) {
-                total_transform += " translate(" + this.x_offset + ", " +
-                    this.y_offset + ")";
+        get_rotation: function() {
+            var rotate_angle = this.model.get("rotate_angle");
+            var transform = "";
+            if(rotate_angle !== undefined) {
+                transform += " rotate(" + rotate_angle + ")";
             }
-
-            if(this.rotate_angle !== undefined) {
-                total_transform += " rotate(" + this.rotate_angle + ")";
-            }
-
-            return total_transform;
+            return transform;
         },
-        apply_net_transform: function() {
-            // this function gets the net transform after applying both the
-            // rotate and x, y trasnforms
+        update_position: function() {
             var x = (this.x_scale.model.type === "date") ?
                 this.model.get_date_elem("x") : this.model.get("x");
             var y = (this.y_scale.model.type === "date") ?
                 this.model.get_date_elem("y") : this.model.get("y");
-            var net_transform = "translate(" + (this.x_scale.scale(x) + this.x_scale.offset) +
-                ", " + (this.y_scale.scale(y) + this.y_scale.offset) +
-                ")";
-            net_transform += this.get_extra_transform();
-            this.el.selectAll(".label")
-                .attr("transform", net_transform);
+            var x_offset = this.model.get("x_offset"),
+                y_offset = this.model.get("y_offset");
+            this.el.select(".label")
+                .attr("transform", "translate(" + 
+                    (this.x_scale.scale(x) + this.x_scale.offset + x_offset) + "," +
+                    (this.y_scale.scale(y) + this.y_scale.offset + y_offset) + ")" +
+                    this.get_rotation());
         },
         update_text: function(model, value) {
-            this.text = value;
             this.el.select(".label")
-                .text(this.text);
+                .text(value);
         },
         update_style: function() {
-            this.color = this.model.get("color");
             this.el.select(".label")
                 .style("font-size", this.model.get("font_size"))
                 .style("font-weight", this.model.get("font_weight"))
                 .style("text-anchor", this.model.get("align"));
-
-            if(this.color !== undefined) {
+            
+            var color = this.model.get("color");
+            if(color !== undefined) {
                 this.el.select(".label")
-                    .style("fill", this.color);
+                    .style("fill", color);
             }
         },
+        set_drag_behavior: function() {
+            var label = this.el.select(".label");
+            if (this.model.get("enable_move")) {
+                label.call(this.drag_listener)
+                    .style("cursor", "move");
```

Comment:
im not sure about this. no other `drag` functionality has a cursor. Typically, we make the element larger, but if we move all drag to this I'm okay with this change. Let's keep it for another PR.